### PR TITLE
cs: adds logging for the iteration where the rebalance was a no-op

### PR DESCRIPTION
**Pull request by shamsimam**
## Changes proposed in this PR

- log when the rebalance was a no-op
- minor refactoring to sort keys and remove unused values

## Why are we making these changes?

Helps us quickly identify why a rebalance iteration was a no-op.

### CODE REVIEW

File: scheduler/src/cook/mesos/rebalancer.clj [hunk 1]
```diff
@@ -538,19 +538,22 @@
                                                  (select-keys (keywordize-keys (:resources v))
                                                               [:cpus :mem :gpus])]))
                                          (into {}))]
-          (when (and (seq params)
-                     (> utilization (:min-utilization-threshold params)))
+          (if (and (seq params)
+                   (> utilization (:min-utilization-threshold params)))
```
Comment (dposada) [inline scheduler/src/cook/mesos/rebalancer.clj]: Can we destructure this?

File: scheduler/src/cook/mesos/rebalancer.clj [hunk 2]
```diff
@@ -538,19 +538,22 @@
...
+                                        :compute-pending-job-dru compute-pending-normal-job-dru))
               (rebalance! conn driver offer-cache gpu-pending-jobs host->spare-resources
                           rebalancer-reservation-atom
-                          (assoc params :compute-pending-job-dru compute-pending-gpu-job-dru
-                                        :category :gpu))))))
-        {:error-handler (fn [ex] (log/error ex "Rebalance failed"))})
-      #(async/close! trigger-chan)))
+                          (assoc params :category :gpu
+                                        :compute-pending-job-dru compute-pending-gpu-job-dru)))
+            (log/info "Rebalance iteration is a no-op"
```
Comment (dposada) [inline scheduler/src/cook/mesos/rebalancer.clj]: Maybe "Skipping rebalancing due to low cluster utilization"?

File: scheduler/src/cook/mesos/rebalancer.clj (hunk 1 above)
Comment (shamsimam) [reply scheduler/src/cook/mesos/rebalancer.clj]: Done.

File: scheduler/src/cook/mesos/rebalancer.clj (hunk 2 above)
Comment (shamsimam) [reply scheduler/src/cook/mesos/rebalancer.clj]: That may not be true as `min-utilization-threshold` may be nil.

**Comment by shamsimam** [comment]
@dposada build is green, can we merge?

File: scheduler/src/cook/mesos/rebalancer.clj (hunk 2 above)
Comment (pschorf) [reply scheduler/src/cook/mesos/rebalancer.clj]: Should we just disable the rebalancer in that case? I like @dposada 's message better.

Comment (shamsimam) [reply scheduler/src/cook/mesos/rebalancer.clj]: We can't disable the rebalancer as the data might be written to the DB after Cook has been running for a while. I'll change the messaging.

Comment (shamsimam) [reply scheduler/src/cook/mesos/rebalancer.clj]: Changed the log message.

**Comment by dposada** [comment]
I am good with merging. Should we wait for @pschorf to approve?

**Pull request merged**